### PR TITLE
[feature] implement erc20 get-transaction for SNT and STT

### PR DESCRIPTION
**Pull request by yenda**
- change `ethereum/get-transaction` to use call-private-rpc instead
of web3 object
- implement a version for erc20 tokens SNT and STT that provides enriches
result of get-transaction with `:recipient` and `:snt-amount`

### CODE REVIEW

File: src/status_im/utils/ethereum/abi_spec.cljs [hunk 1]
```diff
@@ -41,7 +41,7 @@
 
 (defn hex-to-number [x]
   (when x
-    (.toNumber (dependencies/Web3.prototype.toBigNumber (str "0x" x) 16))))
+    (dependencies/Web3.prototype.toBigNumber (str "0x" x) 16)))
```
Comment (flexsurfer) [inline src/status_im/utils/ethereum/abi_spec.cljs]: `.toBigNumber` has only one parameter, so 16 is just confusing and better to remove it
how does it work without `.toNumber`?

File: src/status_im/utils/ethereum/core.cljs [hunk 2]
```diff
@@ -175,11 +176,31 @@
                                      (cb (js->clj result :keywordize-keys true))
                                      (handle-error error)))))
 
-(defn get-transaction [web3 number cb]
-  (.getTransaction (.-eth web3) number (fn [error result]
-                                         (if-not error
-                                           (cb (js->clj result :keywordize-keys true))
-                                           (handle-error error)))))
+#_(defn get-transaction [web3 number cb]
```
Comment (flexsurfer) [inline src/status_im/utils/ethereum/core.cljs]: remove?

**Review by flexsurfer — APPROVED** [review]

File: src/status_im/utils/ethereum/abi_spec.cljs (hunk 1 above)
Comment (flexsurfer) [inline src/status_im/utils/ethereum/abi_spec.cljs]: `.toBigNumber` has only one parameter, so better to remove 16 to do not confuse

File: src/status_im/utils/ethereum/core.cljs [hunk 3]
```diff
@@ -175,11 +176,25 @@
...
+  (first (abi-spec/decode hex ["uint"])))
+
+(defn get-transaction [transaction-hash callback]
+  (status/call-private-rpc
+   (.stringify js/JSON (clj->js {:jsonprc "2.0"
+                                 :id      1
+                                 :method  "eth_getTransactionByHash"
+                                 :params  [transaction-hash]}))
+   (fn [response]
+     (if (= "" response)
```
Comment (vitvly) [inline src/status_im/utils/ethereum/core.cljs]: Can we use string/blank? instead?

**Review by vitvly — CHANGES_REQUESTED** [review]

**Review by vitvly — APPROVED** [review]

**Comment by oskarth** [comment]
@yenda what's the state of this PR?

**Comment by yenda** [comment]
It doesn't need to be tested just e2e tests as there is no changes to previous code

**Comment by churik** [comment]
@yenda e2e cannot be built:
```
The triggering condition was at timestamp=1556005729: inotify-add-watch(/home/jenkins/workspace/us-react_prs_android-e2e_PR-7551/node_modules/unset-value/node_modules/has-value) -> The user limit on the total number of inotify watches was reached; increase the fs.inotify.max_user_watches sysctl
09:48:54  All requests will continue to fail with this message until you resolve
09:48:54  the underlying problem.  You will find more information on fixing this at
09:48:54  https://facebook.github.io/watchman/docs/troubleshooting.html#poison-inotify-add-watch. Make sure watchman is running for this project. See https://facebook.github.io/watchman/docs/troubleshooting.html.
```
Infrastructure error?

**Pull request merged**